Kinvey/Kinvey/DeviceInfo.swift

```diff
@@ -0,0 +1,205 @@
+//
+//  DeviceInfo.swift
+//  Kinvey
+//
+//  Created by Victor Hugo Carvalho Barros on 2018-04-26.
+//  Copyright © 2018 Kinvey. All rights reserved.
+//
+
+import Foundation
+
+#if canImport(UIKit)
+    import UIKit
+    #if canImport(WatchKit)
+        import WatchKit
+    #endif
+#elseif canImport(Cocoa)
+    import Cocoa
+#endif
+
+struct DeviceInfo: Codable {
+    
+    let version: Int
+    let model: String
+    
+    let operationSystem: String
+    let operationSystemVersion: String
+    
+    let platform: String
+    let platformVersion: String
+    
+    let deviceType: String?
+    let netorkCondition: String?
+    let deviceId: String?
+    
+    static var platform: String {
+        #if os(iOS)
+            return "iOS"
+        #elseif os(tvOS)
+            return "tvOS"
+        #elseif os(watchOS)
+            return "watchOS"
+        #elseif os(macOS)
+            return "macOS"
+        #elseif os(Linux)
```

Comment:
Would the platform ever be Linux?

Reply:
well, Swift can run in Linux, but currently we are not supporting Linux for now. Maybe we can add this to the roadmap.